src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -1387,38 +1427,40 @@ MapArchiveResult MetaspaceShared::map_archives(FileMapInfo* static_mapinfo, File
   if (result == MAP_ARCHIVE_SUCCESS) {
     SharedBaseAddress = (size_t)mapped_base_address;
 #ifdef _LP64
-        if (Metaspace::using_class_space()) {
-          // Set up ccs in metaspace.
-          Metaspace::initialize_class_space(class_space_rs);
-
-          // Set up compressed Klass pointer encoding: the encoding range must
-          //  cover both archive and class space.
-          address cds_base = (address)static_mapinfo->mapped_base();
-          address ccs_end = (address)class_space_rs.end();
-          assert(ccs_end > cds_base, "Sanity check");
-          if (INCLUDE_CDS_JAVA_HEAP || UseCompactObjectHeaders) {
-            // The CDS archive may contain narrow Klass IDs that were precomputed at archive generation time:
-            // - every archived java object header (only if INCLUDE_CDS_JAVA_HEAP)
-            // - every archived Klass' prototype   (only if +UseCompactObjectHeaders)
-            //
-            // In order for those IDs to still be valid, we need to dictate base and shift: base should be the
-            // mapping start, shift the shift used at archive generation time.
-            address precomputed_narrow_klass_base = cds_base;
-            const int precomputed_narrow_klass_shift = ArchiveBuilder::precomputed_narrow_klass_shift();
-            CompressedKlassPointers::initialize_for_given_encoding(
-              cds_base, ccs_end - cds_base, // Klass range
-              precomputed_narrow_klass_base, precomputed_narrow_klass_shift // precomputed encoding, see ArchiveBuilder
-            );
-          } else {
-            // Let JVM freely chose encoding base and shift
-            CompressedKlassPointers::initialize (
-              cds_base, ccs_end - cds_base // Klass range
-              );
-          }
-          // map_or_load_heap_region() compares the current narrow oop and klass encodings
-          // with the archived ones, so it must be done after all encodings are determined.
-          static_mapinfo->map_or_load_heap_region();
-        }
+    if (Metaspace::using_class_space()) {
+      assert(prot_zone_size > 0 &&
```

Comment:
This code assumes that `prot_zone_size > 0`, but we have other code that checks `if (prot_zone_size > 0)`. Should the "if" be changed to asserts?